File: webconsole/src/main/resources/_webapp/src/app/charts/model/axis/resource.name.axis.ts
import { Axis } from './abstract.axis';

export class ResourceNameAxis extends Axis {
    public type:string = Axis.RESOURCE;
    constructor() {
        super();
        this.name = "resources";
    }

    public toJSON():any {
        let _value:any = {};
        _value["@type"] = this.type;
        _value["name"] = this.name;
        return _value;
    }
}

File: webconsole/src/main/java/com/bytex/snamp/web/serviceModel/charts/ResourceNameAxis.java
package com.bytex.snamp.web.serviceModel.charts;

import org.codehaus.jackson.annotate.JsonIgnore;
import org.codehaus.jackson.annotate.JsonTypeName;

/**
 * Represents axis with instance names.
 * @author Roman Sakno
 * @version 2.0
 * @since 2.0
 */
@JsonTypeName("resource")
public final class ResourceNameAxis extends Axis {
    public ResourceNameAxis(){
        setName("resources");
    }

    @Override
    @JsonIgnore
    public String getUOM() {
        return "";
    }
}
